src/app/pages/component-viewer/component-viewer.ts

```diff
@@ -23,7 +23,9 @@ export class ComponentViewer {
               public _componentPageTitle: ComponentPageTitle,
               public docItems: DocumentationItems) {
     this._route.params.subscribe(params => {
-      this.componentDocItem = docItems.getItemById(params['id']);
+      this.componentDocItem = docItems.getItemById(
+          params['id'],
+          this._route.parent.snapshot.params['section']);
```

Comment:
Maybe?

```ts
Observable.combineLatest(this._route.params.pluck('id'), this._route.parent.params.pluck('section'))
  .map(([id, section]) => docItems.getItemById(id, section))
  .takeUntil(this._onDestroy)
  .subscribe(docItem => ...);
```

Reply:
Good suggestion; the typings for `pluck` are messy, so I did it manually from the tuple type
```ts
    Observable.combineLatest(_route.params, _route.parent.params)
        .map((p: [Params, Params]) => ({id: p[0]['id'], section: p[1]['section']}))
        .map(p => docItems.getItemById(p.id, p.section))
        .subscribe(d => {...});
```